src/io/parquet.rs

```diff
@@ -0,0 +1,173 @@
+use super::file_appender::FileAppender;
+use anyhow::Result;
+use arrow::record_batch::RecordBatch;
+use arrow_schema::SchemaRef;
+use async_trait::async_trait;
+use bytes::Bytes;
+use futures::future::BoxFuture;
+use opendal::Operator;
+use parquet::arrow::AsyncArrowWriter;
+use std::{mem, task::Poll};
+use tokio::io::AsyncWrite;
+
+pub struct ParquetAppender {
+    writer: AsyncArrowWriter<OperatorWrapper>,
+}
+
+enum OperatorWrapper {
+    Idle(Operator, String),
+    Poll(
+        Operator,
+        String,
+        BoxFuture<'static, std::result::Result<(), std::io::Error>>,
+        usize,
+    ),
+    Empty,
+}
+
+impl OperatorWrapper {
+    pub fn new(operator: Operator, path: String) -> Self {
+        OperatorWrapper::Idle(operator, path)
+    }
+}
+
+impl AsyncWrite for OperatorWrapper {
+    fn poll_write(
+        mut self: std::pin::Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+        buf: &[u8],
+    ) -> std::task::Poll<std::result::Result<usize, std::io::Error>> {
+        let mut res = None;
+        *self = match mem::replace(self.as_mut().get_mut(), OperatorWrapper::Empty) {
+            OperatorWrapper::Idle(operator, path) => {
+                // Wake up to make sure the future is ready after the
+                // future has been built.
+                cx.waker().wake_by_ref();
+                let opertor_clone = operator.clone();
+                let path_clone = path.clone();
+                let bytes = Bytes::copy_from_slice(buf);
+                let fut: BoxFuture<'static, std::result::Result<(), std::io::Error>> =
+                    Box::pin(async move {
+                        opertor_clone.write(&path_clone, bytes).await?;
```

Comment:
`op.write` will overwrite the path every time. We need to work of `opendal::Writer` instead.